packages/@aws-cdk/aws-vpcv2-alpha/lib/ipam.ts

```diff
@@ -0,0 +1,537 @@
+ * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-ipamscope.html
+ */
+  public readonly publicScope: IIpamScopeBase;
+
+  /**
+   * Provides access to default private IPAM scope through add pool method.
+   * Usage: To add an Ipam Pool to a default private scope
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-ipamscope.html
+   * */
+  public readonly privateScope: IIpamScopeBase;
+
+  // Resource IPAM
+  private readonly _ipam: CfnIPAM;
+  /**
+   * Access to Ipam resource id that can be used later to add a custom private scope to this IPAM
+   * @attribute IpamId
+   */
+  public readonly ipamId: string;
+
+  /**
+   * List of operating regions for IPAM
+   */
+  public readonly operatingRegions: string[];
+
+  /**
+   * List of scopes created under this IPAM
+   */
+  public readonly scopes: IIpamScopeBase[] = [];
+
+  constructor(scope: Construct, id: string, props?: IpamProps) {
+    super(scope, id);
+
+    this.operatingRegions = props?.operatingRegion ?? [Stack.of(this).region];
+
+    this._ipam = new CfnIPAM(this, 'Ipam', {
+      operatingRegions: this.operatingRegions ? this.operatingRegions.map(region => ({ regionName: region })) : [],
+    });
+    this.ipamId = this._ipam.attrIpamId;
+    this.publicScope = new IpamScopeBase(this, 'DefaultPublicScope', {
+      ipamOperatingRegions: this.operatingRegions,
+      ipamId: this._ipam.attrIpamId,
+      ipamScopeId: this._ipam.attrPublicDefaultScopeId,
+    });
+    this.privateScope = new IpamScopeBase(this, 'DefaultPrivateScope', {
+      ipamOperatingRegions: this.operatingRegions,
+      ipamId: this._ipam.attrIpamId,
+      ipamScopeId: this._ipam.attrPrivateDefaultScopeId,
+    });
+
+    this.scopes.push(this.publicScope, this.privateScope);
+    this.scopes.forEach(node => {
+      this.node.defaultChild = node.scope;
+    });
+  }
+
+  /**
+   * Function to add custom scope to an existing IPAM
+   * Custom scopes can only be private
+   */
+  public addScope(scope: Construct, id: string, options: IpamScopeOptions): IIpamScopeBase {
+    const ipamScope = new IpamScope(scope, id, {
+      ...options,
+      ipamId: this.ipamId,
+      ipamOperatingRegions: this.operatingRegions,
+    });
+    this.scopes.push(ipamScope);
+    return ipamScope;
+  }
+}
+
+/**
+ * Function to create IpamPool under scope
+ * @internal
+ */
+function createIpamPool(
+  scope: Construct,
+  id: string,
+  scopeOptions: IpamScopeProps,
+  poolOptions: PoolOptions,
+  scopeId: string,
+): IpamPool {
+  const isLocaleInOperatingRegions = scopeOptions.ipamOperatingRegions
+    ? scopeOptions.ipamOperatingRegions.map(region => ({ regionName: region }))
+      .some(region => region.regionName === poolOptions.locale)
+    : false;
+
+  if (!isLocaleInOperatingRegions) {
+    throw new Error(`The provided locale '${poolOptions.locale}' is not in the operating regions.`);
+  }
```

Comment:
[blocking] should this error thrown even if Operating Regions is an empty list or null?

Reply:
Operating regions cannot be empty list or null, by default it will be set to stack region if not provided, and we throw out an error in IPAM class if both stack region and operating regions are set to null by the user, so this will always has some value. I realized i didn't add this validation, fixed in latest commit here